src/main/java/com/openshift/internal/restclient/model/KubernetesResource.java

```diff
@@ -161,20 +158,26 @@ public void addLabel(String key, String value) {
 	
 	/*---------- utility methods ------*/
 	protected ModelNode get(String key){
-		String [] property = propertyKeys.get(key);
-		return node.get(property);
+		return node.get(getPath(key));
 	}
-	
+
 	protected void set(String key, int value) {
-		String [] property = propertyKeys.get(key);
-		node.get(property).set(value);
+		node.get(getPath(key)).set(value);
 	}
 	
 	protected void set(String key, String value){
-		String [] property = propertyKeys.get(key);
-		node.get(property).set(value);
+		node.get(getPath(key)).set(value);
 	}
 	
+	private String[] getPath(String key) {
+		String [] property = propertyKeys.get(key);
+		if (property == null) {
```

Comment:
@adietish we should consider throwing here/providing an indication that the path doesnt exist.  If the path doesnt exist for a given property then it needs to be added to the map IMO.